tests/integration/devfile/cmd_devfile_init_test.go

```diff
@@ -46,6 +56,19 @@ var _ = Describe("odo devfile init command tests", func() {
 		})
 	})
 
+	When("running odo init from a directory with sources", func() {
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+		})
+		It("should work without --starter flag", func() {
+			helper.Cmd("odo", "init", "--name", "aname", "--devfile", "go").ShouldPass()
```

Comment:
```suggestion
			helper.Cmd("odo", "init", "--name", "aname", "--devfile", "nodejs").ShouldPass()
```
Or was using a Go devfile in a nodejs project intentional?